dask/array/core.py

```diff
@@ -1384,6 +1384,15 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
                         "chunk_type": typename(type(self._meta)),
                     }
                 )
+        if (
+            config.get("array.automatic-rechunk")
+            and any(len(c) > 1 for c in self._chunks)
+            and reduce(mul, map(max, self._chunks))
+            < parse_bytes(config.get("array.chunk-size"))
+            / 2
```

Comment:
IIUC this would only kick in if we're below 50% of the recommend size, not every single time

Reply:
Yes correct

Reply:
yes, agreed